### PR TITLE
FLINK-30199: Add a script to run Kubernetes Operator e2e tests manually

### PR DESCRIPTION
**Pull request by pvary**
## What is the purpose of the change

Adding a possibility to run the e2e tests manually and using this same script to run the tests on the CI

## Brief change log

- Refactored the test steps to `functions` in the util.sh
- Changed the `ci.yml` to use these new fuctions
- Created a shell script to run the tests

## Verifying this change

- This change is already covered by existing e2e - will see the test results
- Also manually executed the test script

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: no
  - Core observer or reconciler logic that is regularly executed: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? There is a help text in the script

### CODE REVIEW

**Comment by mbalassi** [comment]
cc @wangyang0918 for your information @pvary has reworked the test suite to block tests together on the same container runs, we hope this effectively will give us more CI bandwidth. The minikube instances are reused between the tests, the operator itself is always reinstalled to cleanly separate the logs. I am happy to review, just fyi. 😏

File: e2e-tests/run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+ROOT_DIR=$(dirname $(dirname "$(readlink -f "$0")"))
+source "${ROOT_DIR}/e2e-tests/utils.sh"
```
[the remark this answers is not in the document]
Comment (mbalassi) [reply e2e-tests/run_tests.sh:21]: This is due to the `readlink` implementation on my OSX. I missed this error in the log:
`readlink: illegal option -- f`
There are a couple suggestions here:
https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac
WDYT @pvary?

**Review by mbalassi — COMMENTED** [review]
Awesome @pvary, I added some comments that I would like to ask you to consider before merging.

File: e2e-tests/utils.sh [hunk 2]
```diff
@@ -265,3 +270,75 @@ function create_namespace() {
...
+}
+
+function install_operator() {
+  local namespace=$1
+
+  create_namespace flink
+  if [[ -n ${DEBUG} ]]; then
+    debug="--debug"
+  fi
+  helm ${debug} install flink-kubernetes-operator -n ${namespace} helm/flink-kubernetes-operator --set image.repository=flink-kubernetes-operator --set image.tag=ci-latest --create-namespace --set 'watchNamespaces={default,flink}'
```
[the remark this answers is not in the document]
Comment (pvary) [reply e2e-tests/utils.sh:299]: The namespaces were test parameters even before this change, and the watched namespaces were only an extra param which did not change between the tests.

I tried to keep the existing testing behaviour in this change, and especially do not want to change the tests themselves, so we can be sure that the PR does not cause any unwanted changes.

We might want to do this in a different PR, but even there, I am not sure if it is worth it. I think we gain more by keeping the tests simple and easy to read.

**Review by mbalassi — APPROVED** [review]
Thanks, great work.

**Comment by gaborgsomogyi** [comment]
This change is a great help, thanks guys! 👍

**Pull request merged**